cmd/root.go

```diff
@@ -45,35 +45,58 @@ import (
 	"go.k6.io/k6/log"
 )
 
-//TODO: remove these global variables
-//nolint:gochecknoglobals
-var (
-	outMutex   = &sync.Mutex{}
-	isDumbTerm = os.Getenv("TERM") == "dumb"
-	stdoutTTY  = !isDumbTerm && (isatty.IsTerminal(os.Stdout.Fd()) || isatty.IsCygwinTerminal(os.Stdout.Fd()))
-	stderrTTY  = !isDumbTerm && (isatty.IsTerminal(os.Stderr.Fd()) || isatty.IsCygwinTerminal(os.Stderr.Fd()))
-	stdout     = &consoleWriter{colorable.NewColorableStdout(), stdoutTTY, outMutex, nil}
-	stderr     = &consoleWriter{colorable.NewColorableStderr(), stderrTTY, outMutex, nil}
-)
-
 const (
 	defaultConfigFileName   = "config.json"
 	waitRemoteLoggerTimeout = time.Second * 5
 )
 
-//TODO: remove these global variables
-//nolint:gochecknoglobals
-var defaultConfigFilePath = defaultConfigFileName // Updated with the user's config folder in the init() function below
-//nolint:gochecknoglobals
-var configFilePath = os.Getenv("K6_CONFIG") // Overridden by `-c`/`--config` flag!
-
-//nolint:gochecknoglobals
-var (
-	// TODO: have environment variables for configuring these? hopefully after we move away from global vars though...
-	quiet   bool
-	noColor bool
-	address string
-)
+// TODO better name - there are other commadn flags these are just ... non lib.Options ones :shrug:
+type commandFlags struct {
+	defaultConfigFilePath string
+	configFilePath        string
+	exitOnRunning         bool
+	showCloudLogs         bool
+	runType               string
+	archiveOut            string
+	quiet                 bool
+	noColor               bool
+	address               string
+	outMutex              *sync.Mutex
+	stdoutTTY, stderrTTY  bool
+	stdout, stderr        *consoleWriter
+}
+
+func newCommandFlags() *commandFlags {
+	confDir, err := os.UserConfigDir()
+	if err != nil {
+		logrus.WithError(err).Warn("could not get config directory")
+		confDir = ".config"
+	}
+	defaultConfigFilePath := filepath.Join(
+		confDir,
+		"loadimpact",
+		"k6",
+		defaultConfigFileName,
+	)
+
+	isDumbTerm := os.Getenv("TERM") == "dumb"
+	stdoutTTY := !isDumbTerm && (isatty.IsTerminal(os.Stdout.Fd()) || isatty.IsCygwinTerminal(os.Stdout.Fd()))
+	stderrTTY := !isDumbTerm && (isatty.IsTerminal(os.Stderr.Fd()) || isatty.IsCygwinTerminal(os.Stderr.Fd()))
+	outMutex := &sync.Mutex{}
+	return &commandFlags{
+		defaultConfigFilePath: defaultConfigFilePath,  // Updated with the user's config folder in the init() function below
+		configFilePath:        os.Getenv("K6_CONFIG"), // Overridden by `-c`/`--config` flag!
+		exitOnRunning:         os.Getenv("K6_EXIT_ON_RUNNING") != "",
+		showCloudLogs:         true,
+		runType:               os.Getenv("K6_TYPE"),
```

Comment:
All of those os.Getenv should go, but as currently this are the only places where they are and there are zero tests covering this I decided against introducing even *more* changes.